plugins/net_plugin/net_plugin.cpp

```diff
@@ -1192,16 +1190,21 @@ namespace eosio {
    };
    
 
-   std::tuple<std::string, std::string, std::string> split_host_port_type(const std::string& peer_add) {
+   std::tuple<std::string, std::string, std::string> split_host_port_type(const std::string& peer_add, bool incoming) {
       // host:port:[<trx>|<blk>]
       if (peer_add.empty()) return {};
 
       string::size_type p = peer_add[0] == '[' ? peer_add.find(']') : 0;
-      if (p == string::npos) {
-         fc_wlog( logger, "Invalid peer address: ${peer}", ("peer", peer_add) );
+      string::size_type colon = p != string::npos ? peer_add.find(':', p) : string::npos;
+      if (colon == std::string::npos || colon == 0) {
+         // if incoming then not an error this peer can do anything about
+         if (incoming) {
+            fc_dlog( logger, "Invalid peer address. must be \"host:port[:<blk>|<trx>]\": ${p}", ("p", peer_add) );
+         } else {
+            fc_elog( logger, "Invalid peer address. must be \"host:port[:<blk>|<trx>]\": ${p}", ("p", peer_add) );
+         }
          return {};
```

Comment:
If we return a tuple with 3 empty strings, won't `connection::set_connection_type()`  go into `if( type.empty() ) {` and log that we are setting a connection?

Reply:
It will log: `Setting connection - ${c} type for: ${peer} to both transactions and blocks`. That sound not be possible as peer address should be validated by then. But even if it has not been validated, it logging that it is setting the connection type to `both` is correct. Now if that `both` can never be realized because the address is later determined to be invalid, no harm done.